### PR TITLE
[LLVM][rtsan] Add LLVM attribute sanitize_realtime_unsafe

### PR DESCRIPTION
**Pull request by davidtrevelyan**

### CODE REVIEW

File: llvm/include/llvm/IR/Attributes.td [hunk 1]
```diff
@@ -303,6 +303,10 @@ def SanitizeNumericalStability : EnumAttr<"sanitize_numerical_stability", [FnAtt
 /// RealtimeSanitizer is on.
 def SanitizeRealtime : EnumAttr<"sanitize_realtime", [FnAttr]>;
 
+/// RealtimeSanitizer should error if an realtime_unsafe function is called
+/// during a sanitize_realtime function.
```
Comment (cjappl) [inline llvm/include/llvm/IR/Attributes.td]: During a sanitize_realtime function, or invoked below one

Perhaps something along the lines of:

"RealtimeSanitizer should error if invoked in a realtime context" -- idk documentation is hard.

File: llvm/test/Bitcode/attributes.ll [hunk 2]
```diff
@@ -613,6 +618,7 @@ define void @initializes(ptr initializes((-4, 0), (4, 8)) %a) {
 ; CHECK: attributes #52 = { nosanitize_bounds }
 ; CHECK: attributes #53 = { sanitize_realtime }
 ; CHECK: attributes #54 = { nosanitize_realtime }
```
Comment (cjappl) [inline llvm/test/Bitcode/attributes.ll]: Note that when we strip out this attribute, all of your function "numbers" will be affected, I will try to get this merged asap

**Review by cjappl — COMMENTED** [review]
Overall this looks good. 

you are missing a test to confirm that these attributes are not compatible:
llvm/test/Verifier/rtsan-attrs.ll

and the verifier code to enforce they are not compatible:
llvm/lib/IR/Verifier.cpp

Also, this code will be somewhat affected when I revert the nosanitize attribute

I would also make sure you run git-clang-tidy main, to ensure everything is clean. That is os one of the things that gets me most often.

**Comment by davidtrevelyan** [comment]
Super, thanks @cjappl - will update all of these when the nosanitize_realtime attribute is removed :)

File: llvm/include/llvm/IR/Attributes.td (hunk 1 above)
Comment (davidtrevelyan) [reply llvm/include/llvm/IR/Attributes.td]: Done

File: llvm/test/Bitcode/attributes.ll (hunk 2 above)
Comment (davidtrevelyan) [reply llvm/test/Bitcode/attributes.ll]: Thanks - updated as part of rebasing

**Pull request closed without merging**